physical/raft/raft.go

```diff
@@ -556,7 +556,16 @@ func (b *RaftBackend) SetupCluster(ctx context.Context, opts SetupOpts) error {
 			return errwrap.Wrapf("raft recovery failed to parse peers.json: {{err}}", err)
 		}
 
-		b.logger.Info("raft recovery: found new config", "config", recoveryConfig)
+		// Non-voting servers are only allowed in enterprise. If Suffage is disabled,
+		// error out to indicate that it isn't allowed.
+		for idx := range recoveryConfig.Servers {
+			if !nonVotersAllowed && recoveryConfig.Servers[idx].Suffrage == raft.Nonvoter {
+				return fmt.Errorf("raft recovery failed: `non_voter` is only supported in enterprise")
```

Comment:
Not a huge deal since the file is relatively small and easy to read, but if multiple peers were provided it might be helpful to also return the Node ID for which this was set to true.

Reply:
Done